comms/dht/src/peer_validator.rs

```diff
@@ -73,9 +76,23 @@ impl<'a> PeerValidator<'a> {
             });
         }
 
+        if let Some(existing) = &existing_peer {
+            if existing.public_key != new_peer.public_key {
+                return Err(DhtPeerValidatorError::NewAndExistingMismatch {
+                    existing: format!("'{}' / '{}'", existing.node_id.to_hex(), existing.public_key.to_hex()),
```

Comment:
To hex strings (2 allocations) are not required. Can do alloc free display with:

```suggestion
                    existing: format!("'{}' / '{}'", existing.node_id, existing.public_key),
```